cmd/zoekt-indexserver/main.go

```diff
@@ -164,6 +166,61 @@ func indexPendingRepos(indexDir, repoDir string, opts *Options, repos <-chan str
 	}
 }
 
+type indexRequest struct {
+	RepositoryUrl  string // TODO: Decide if tokens can be in the URL or if we should pass separately
```

Comment:
I may initially start with public project support only but we do need some way to index private repos. I noticed the current GitLab indexer doesn't pass the token through so I had a separate change for that https://github.com/sourcegraph/zoekt/commit/18b83db289931039cb85503290c46e19816fe4a5 but this doesn't use the gitlab indexer. I also had concerns about adding the token to the URL and how that might end up being logged all over the place. I didn't want to tackle this problem at the same time as the indexing API but the design will need to account for it.